pds4indextools/pds4_create_collection_index.py

```diff
@@ -0,0 +1,81 @@
+"""
+PDS4 Collection Product Creation Tool
+
+This script scrapes label files within a collection for their identifying information,
+and creates a collection product file containing the LIDVIDs and the Member Status of the
+label files. This tool can also generate the collection product elsewhere within the
+user's directory, for use during the creation of collections. For the full instructions
+on how to use the tool, see:
+
+python pds4_create_collection_product.py --help
+"""
+
+import argparse
+from lxml import etree
+from pathlib import Path
+import csv
+
+
+def main():
+
+    # Parse command line arguments
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('collectionpath', type=str, metavar='COLLECTION_PATH',
+                        help='The path to the collection product')
+    parser.add_argument('bundle', type=str, metavar='BUNDLE_NAME',
+                        help='The name of the bundle')
+    parser.add_argument('collection', type=str, metavar='COLLECTION_NAME',
+                        help='The name of the collection')
+    parser.add_argument('--collection-product-file', type=str,
+                        metavar='COLLECTION_PRODUCT_FILEPATH',
+                        help='The output location of the collection product.')
+
+    args = parser.parse_args()
+
+    collection_path = Path(args.collectionpath)
+    label_files = []
+    primary = f"{args.bundle}:{args.collection}"
+
+    # Gather all XML files in the collection path, sort by directory structure
+    label_files = collection_path.glob('**/*.xml')
+    label_files = sorted(label_files, key=lambda x: x.parent)
```

Comment:
Just to be clear, this sort does automatically exhaust the iterator, meaning the glob will run to completion. For a large directory structure this could take some time. But if you want to sort on directory/filename, there really isn't any other choice.

I'm a little confused by the sort key, though. Isn't the parent the directory containing the file? So what happens to the sort order within a particular directory. It seems like that is unconstrained and will end up being random.